### ISSUE TITLE
Templates and css-ID/Class for Include-Elements

### ISSUE THREAD

**Issue by 463** [contao/contao#5752]
### Description

Is there any reason why not all Include-Elements have an individual template like all other content elements?
- Article has neither a template no a css-ID/Class input field. This results in a missing and adjustable covering `<div></div>`
- Content-Element has no template (but it has css-ID/Class)
- Form has no template (but it has css-ID/Class)
- Module has no template (but it has css-ID/Class)
All further have both.

Especially the Article-Element may becomes useless if the included one is not the only in the parent article because there is no chance for css styling this element.

As workaround only manual HTML-elements with `<div class="..." ></div>` are a solution but in this case insert-tags are the more simple solution.

So fare the Article-Element has not really an advantage.

A comparable problem is already discussed years ago for HTML-elements [https://github.com/contao/core/issues/1148](url)

**Comment by fritzmg** [comment]
> Article has neither a template no a css-ID/Class input field. This results in a missing and adjustable covering `<div></div>`

That is intentional. The `article` include element only includes the _content_ of the article, otherwise you would end up with

```html
<div class="mod_article">
  <div class="mod_article">
    …
```

**Comment by 463** [comment]
That's ok. The inner div with `mod_article` must be omitted to avoid a loop.

But the outer div is completely missing. This should get e. g. `ce_article block` as a content element not a module.

**Comment by 463** [comment]
Of course, it can occur a `ce_article` "loop". But that can also happen using `{{insert_article::*}}` inside a `ce_text` element. Such "loop" should be never a problem because this are only css classes without any further hierarchical logic behind.

**Comment by fritzmg** [comment]
> That's ok. The inner div with `mod_article` must be omitted to avoid a loop.
> 
> But the outer div is completely missing. This should get e. g. `ce_article block` as a content element not a module.

Not sure what you mean. There is no outer or inner div for articles. There is only `<div class="mod_article">` - which is not in use when using the article include element for aforementioned reasons and thus there is also no CSS ID/class input.

**Comment by 463** [comment]
I make an example:

The basis article (that what will be included) is given by

![grafik](https://user-images.githubusercontent.com/7769231/216457646-74fa96d3-4210-4c57-8bce-6a4968ba0331.png)

results in
 
![grafik](https://user-images.githubusercontent.com/7769231/216457939-09445e83-94cf-414b-9ac0-25a18c6faa5b.png)


The including article

![grafik](https://user-images.githubusercontent.com/7769231/216461721-57f96b02-7326-4a9c-8da9-fafe32fd49ae.png)

makes a linear and non-structured output:

![grafik](https://user-images.githubusercontent.com/7769231/216461907-fc312964-0775-4679-beb3-8cd924c322ce.png)

There is no possibility to refer the included article as a single, common block.

Advantageous would be:

```
<div id="article-25" class="mod_article block">
  <div id="article-5-included" class="ce_article block">
    <div class="ce_text block">
      <p>Bla bla 1</p>
    </div>
    <div class="ce_text block">
      <p>Bla bla 2</p>
    </div>
  </div>
  <div class="ce_text block">
    <p>Bla bla 3</p>
  </div>
</div>
```
In this case `id="article-5"` (the basis article) and `id="article-5-included"` (the included article) can be easily referenced together in css `#article-5, #article-5-included {...}` for identical common formatting.

Also ce_article can be used to make an overarching and joint animation in css or js (e. g. a slider).

The proposed `id="article-5-included"` of the included ce_article block is only an example that could possibly be set automatically if no manual ID is set. 

At the end, the div bracket around the complete inserted article is crucial and neither distracting nor colliding with other class names. There is no risk to get a format destruction if not a `#article-25 > .ce_text` or even a `#article-25 + div` or a similar level dependent selector is used.

Alternatively / additionally at least an individual template should be usable where the div bracket can be inserted or deleted (depending on a standard case) to have the optional chance to get this more flexible or the previous behavior.

**Comment by fritzmg** [comment]
> makes a linear and non-structured output:

That's intentional.

> Advantageous would be:

That's a completely different use case. The purpose of the article include element is to include the contents of an article within another article - as if the content was simply defined in that article. Hence there is no additional wrapper and thus you get the same front end output as if you would place individual content elements. That's the intention behind this content element.

**Comment by 463** [comment]
Therefore it should be a feature. 

At least a customisable template and/(or) a css-Id/Class (possibly controlling the insert of the additional div bracket) would not have an influence to any existing behavior but would be a general advancement to use it for content templates e. g. for addresses, phone numbers etc. to centralize the maintenance. For several include elements this customizing opportunity is already existing and working.

**Comment by fritzmg** [comment]
Last time we discussed the future of content elements and articles and may be in the future there won't be this article & content element system anymore. But instead content elements can be nested and thus there can be a new content element type that would represent the old "article" with other nested content elements inside it. This would also cover your feature request as you could then simply nest the article include element (which would then just be a content element include element) inside another article content element.

**Comment by frontendschlampe** [comment]
@463 you can use our extension hofff/contao-content which will provide the functionality you want. You can include articles and decide if you want to get an additional `div` or not. You have to use the content element "Content references" and activate the option "Render article container".

**Comment by aschempp** [comment]
https://github.com/contao/contao/pull/6120 would fix this in Contao 5.2 by adding a `ce_article` template. You can then override that template (or add your own `ce_article_xxx`) if you need a wrapping div etc.
